### PR TITLE
Add release process doc

### PR DESCRIPTION
**Pull request by karlkfi**
I wanted to make sure I'm not the only one who knows how to make releases.
Let me know what you think, @sttts

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+## Release Process
+
+1. Modify master branch
```
Comment (sttts) [inline RELEASE.md:3]: ~~Commit, push!~~

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+8. Push docker image
+
+  ```
+  $ make push
+  docker push mesosphere/mesos-slave-dind:0.2.3_mesos-0.24.0_docker-1.7.1_ubuntu-14.04.3
+  ```
+
+
+To create releases with different docker versions, you will need to create a new release, modify the Dockerfile, and make a commit to the branch before you can make the new docker image. This requires knowing the release version before producing an image, which is why it's not automated. Hopefully we wont need to create too many releases with multiple docker versions.
```
Comment (sttts) [inline RELEASE.md]: won't

**Comment by sttts** [comment]
lgtm

**Comment by s-urbaniak** [comment]
afaict lgtm

**Pull request merged**